src/lib/nfs_client/fstab_entry_dialog.rb

```diff
@@ -0,0 +1,403 @@
+      else
+        host = choose_hostname(@hosts)
+        Yast::UI.ChangeWidget(Id(:serverent), :Value, host) if host
+      end
+    end
+
+    def pathent_list_handler
+      server2 = Yast::UI.QueryWidget(Id(:serverent), :Value).to_s
+      v4 = Yast::UI.QueryWidget(Id(:nfs4), :Value)
+
+      if !CheckHostName(server2)
+        Yast::UI.SetFocus(Id(:serverent))
+        return
+      end
+
+      Yast::UI.OpenDialog(
+        Label(
+          # Popup dialog, %1 is a host name
+          Yast::Builtins.sformat(
+            _("Getting directory list for \"%1\"..."),
+            server2
+          )
+        )
+      )
+      dirs = Yast::Nfs.ProbeExports(server2, v4)
+      Yast::UI.CloseDialog
+
+      dir = choose_export(dirs)
+      Yast::UI.ChangeWidget(Id(:pathent), :Value, dir) if dir
+    end
+
+    def browse_handler
+      dir = Yast::UI.QueryWidget(Id(:mountent), :Value)
+      dir = "/" if dir.nil? || dir.empty?
+      # heading for a directory selection dialog
+      dir = Yast::UI.AskForExistingDirectory(dir, _("Select the Mount Point"))
+
+      return if !dir || dir.empty?
+      Yast::UI.ChangeWidget(Id(:mountent), :Value, dir)
+    end
+
+    def ok_handler
+      server = FormatHostnameForFstab(
+        Yast::UI.QueryWidget(Id(:serverent), :Value)
+      )
+      @pth = StripExtraSlash(
+        Yast::UI.QueryWidget(Id(:pathent), :Value)
+      )
+      @mount = StripExtraSlash(
+        Yast::UI.QueryWidget(Id(:mountent), :Value)
+      )
+      @nfs4 = Yast::UI.QueryWidget(Id(:nfs4), :Value)
+      @nfs41 = Yast::UI.QueryWidget(Id(:nfs41), :Value)
+      @options = Yast::UI.QueryWidget(Id(:optionsent), :Value).tr(" ", "")
+      @options = Yast::NfsOptions.set_nfs41(@options, @nfs41)
+
+      options_error = Yast::NfsOptions.validate(@options)
+      if !CheckHostName(server)
+        Yast::UI.SetFocus(Id(:serverent))
+      elsif !CheckPath(@pth)
+        Yast::UI.SetFocus(Id(:pathent))
+      elsif !CheckPath(@mount) || IsMpInFstab(@existing, @mount)
+        Yast::UI.SetFocus(Id(:mountent))
+      elsif options_error.size > 0
+        Yast::Popup.Error(options_error)
+        Yast::UI.SetFocus(Id(:optionsent))
+      else
+        @fstab_ent = {
+          "spec"    => "#{server}:#{@pth}",
+          "file"    => @mount,
+          "vfstype" => @nfs4 ? "nfs4" : "nfs",
+          "mntops"  => @options
+        }
+        @fstab_ent["old"] = @old unless @old == @fstab_ent["spec"]
+        Yast::Wizard.RestoreScreenShotName
+        log.info "FstabEntryDialog returns #{@fstab_ent}"
+        finish_dialog(@fstab_ent)
+      end
+    end
+
+    def help_handler
+      # help text 1/4
+      # change: locally defined -> servers on LAN
+      helptext = _(
+        "<p>Enter the <b>NFS Server Hostname</b>.  With\n" \
+          "<b>Choose</b>, browse through a list of\n" \
+          "NFS servers on the local network.</p>\n"
+      )
+      # help text 2/4
+      # added "Select" button
+      helptext << _(
```

Comment:
well, it crash only if you use non-english as for english it return parameter, but if it return translated string, it is frozen